### PR TITLE
[Dev portal][User management] - Enhance user delete feature

### PR DESCRIPTION
**Pull request by NipuniBhagya**
## Screenshots
image

### CODE REVIEW

File: apps/admin-portal/src/components/users/user-profile.tsx [hunk 1]
```diff
@@ -49,10 +55,14 @@ export const UserProfile: FunctionComponent<ProfileProps> = (props: ProfileProps
...
-    const [profileInfo, setProfileInfo] = useState(new Map<string, string>());
-    const [profileSchema, setProfileSchema] = useState<ProfileSchema[]>();
-    const profileDetails: AuthStateInterface = useSelector((state: AppState) => state.authenticationInformation);
-    const [urlSchema, setUrlSchema] = useState<ProfileSchema>();
+    const [ profileInfo, setProfileInfo ] = useState(new Map<string, string>());
+    const [ profileSchema, setProfileSchema ] = useState<ProfileSchema[]>();
+    const profileDetails: AuthStateInterface = useSelector((state: AppState) =>
+        state.authenticationInformation);
+    const [ urlSchema, setUrlSchema ] = useState<ProfileSchema>();
+
```
Comment (inthirakumaaran) [inline apps/admin-portal/src/components/users/user-profile.tsx]: Shall we remove this empty space or add a comment if you want to separate them

Comment (NipuniBhagya) [reply apps/admin-portal/src/components/users/user-profile.tsx]: I separated them cos they were related to confirmation modal. Should we add a comment though?

File: apps/admin-portal/src/components/users/user-profile.tsx [hunk 2]
```diff
@@ -17,15 +17,21 @@
...
 import { isEmpty } from "lodash";
 import React, { FunctionComponent, ReactElement, useEffect, useState } from "react";
 import { useTranslation } from "react-i18next";
 import { useSelector } from "react-redux";
 import { Button, Divider, Grid } from "semantic-ui-react";
 import { deleteUser, updateUserInfo } from "../../api";
 import { history } from "../../helpers";
-import { AlertInterface, AlertLevels, AuthStateInterface, BasicProfileInterface, ProfileSchema } from "../../models";
+import {
+    AlertInterface,
```
Comment (brionmario) [inline apps/admin-portal/src/components/users/user-profile.tsx]: `AlertInterface` and `AlertLevels` are available in the `@wso2is/core/models`

File: apps/admin-portal/src/components/users/user-profile.tsx [hunk 3]
```diff
@@ -16,16 +16,21 @@
...
 import { useSelector } from "react-redux";
 import { Button, Divider, Grid } from "semantic-ui-react";
 import { deleteUser, updateUserInfo } from "../../api";
 import { history } from "../../helpers";
-import { AlertInterface, AlertLevels, AuthStateInterface, BasicProfileInterface, ProfileSchema } from "../../models";
+import {
+    AuthStateInterface,
+    BasicProfileInterface,
+    ProfileSchema,
+} from "../../models";
```
Comment (jeradrutnam) [inline apps/admin-portal/src/components/users/user-profile.tsx]:
```suggestion
import { AuthStateInterface, BasicProfileInterface, ProfileSchema } from "../../models";
```

File: apps/admin-portal/src/components/users/user-profile.tsx [hunk 4]
```diff
@@ -49,10 +50,13 @@ export const UserProfile: FunctionComponent<ProfileProps> = (props: ProfileProps
...
-    const [profileSchema, setProfileSchema] = useState<ProfileSchema[]>();
-    const profileDetails: AuthStateInterface = useSelector((state: AppState) => state.authenticationInformation);
-    const [urlSchema, setUrlSchema] = useState<ProfileSchema>();
+    const [ profileInfo, setProfileInfo ] = useState(new Map<string, string>());
+    const [ profileSchema, setProfileSchema ] = useState<ProfileSchema[]>();
+    const [ urlSchema, setUrlSchema ] = useState<ProfileSchema>();
+    const [ showDeleteConfirmationModal, setShowDeleteConfirmationModal ] = useState<boolean>(false);
+    const [ deletingUser, setDeletingUser ] = useState<BasicProfileInterface>(undefined);
+    const profileDetails: AuthStateInterface = useSelector((state: AppState) =>
+        state.authenticationInformation);
```
Comment (jeradrutnam) [inline apps/admin-portal/src/components/users/user-profile.tsx]:
```suggestion
    const profileDetails: AuthStateInterface = useSelector((state: AppState) => state.authenticationInformation);
```

**Pull request merged**